src/main/java/com/google/sps/servlets/SignUpServlet.java

```diff
@@ -24,6 +24,24 @@
 @WebServlet("/sign-up")
 public class SignUpServlet extends HttpServlet {
 
+	// Check if user with specified field exists
+	static private boolean checkIfFieldExists(String field, String value) {
+
+		 // Check if value already exists within Datastore
+		 Query<Entity> fieldQuery = Query.newEntityQueryBuilder()
+			 .setKind("User")
+			 .setFilter(PropertyFilter.eq(field, value))
+			 .build();
+		 QueryResults<Entity> entities = datastore.run(fieldQuery);
+		 boolean fieldExists = false;
+
+		 while(entities.hasNext() && !fieldExists){
```

Comment:
Spaces before and after (). 

But do you actually need a loop here? You're not advancing in the loop because you never call next(). Luckily you don't need to do that because if hasNext does not return true immediately that means the field does not exist in the first place. So, your method can just return directly whether entities is NOT null and NOT empty.

Reply:
That makes sense, I will just set fieldExists = entities.hasNext(); that way if there are no entities with that field value it will be false